src/pipeline.ts

```diff
@@ -489,14 +551,483 @@ export async function scanRssFeeds() {
+	let inClient = (await getClient().isTorrentComplete(meta.infoHash)).isOk();
+	let injected = false;
+	for (const [searchee, decision] of [...fullMatches, ...partialMatches]) {
+		const linkedFilesRootResult = await linkAllFilesInMetafile(
+			searchee,
+			meta,
+			tracker,
+			decision,
+			{ onlyCompleted: false },
+		);
+		const linkResult = linkedFilesRootResult.isOk()
+			? linkedFilesRootResult.unwrap()
+			: null;
+		if (linkResult && linkResult.linkedNewFiles) {
+			linkedNewFiles = true;
+		}
+		if (!inClient) {
+			if (linkedFilesRootResult.isOk()) {
+				const destinationDir = dirname(linkResult!.contentPath);
+				const result = await getClient().inject(
+					meta,
+					searchee,
+					Decision.MATCH_PARTIAL, // Should always be considered partial
+					destinationDir,
+				);
+				// result is only SUCCESS or FAILURE here but still log original injectionResult
+				if (result === InjectionResult.SUCCESS) {
+					logger.info({
+						label: Label.INJECT,
+						message: `${progress} Injected ${filePathLog} using stalled source, you will need to resume or remove from client - ${chalk.green(injectionResult)}`,
+					});
+					inClient = true;
+					injected = true;
+				} else {
+					logger.warn({
+						label: Label.INJECT,
+						message: `${progress} Failed to inject ${filePathLog} using stalled source - ${chalk.yellow(injectionResult)}`,
+					});
+				}
+			} else {
+				logger.warn({
+					label: Label.INJECT,
+					message: `${progress} Failed to link files for ${filePathLog}, ${linkedFilesRootResult.unwrapErr()} - ${chalk.yellow(injectionResult)}`,
+				});
+			}
+		}
+	}
+	if (inClient && !injected) {
+		if (linkedNewFiles) {
+			logger.info({
+				label: Label.INJECT,
+				message: `${progress} Rechecking ${filePathLog} as new files were linked - ${chalk.green(injectionResult)}`,
+			});
+			await getClient().recheckTorrent(meta.infoHash);
+		} else {
+			logger.warn({
+				label: Label.INJECT,
+				message: `${progress} No new files linked for ${filePathLog}, resume or remove from client - ${chalk.yellow(injectionResult)}`,
+			});
+		}
+	}
+	return injected;
+}
+
+function injectSummary(summary: InjectSummary): void {
+	const incompleteMsg = `${chalk.bold.yellow(summary.ALREADY_EXISTS)} existed in client${
+		summary.INCOMPLETE_CANDIDATES
+			? chalk.dim(` (${summary.INCOMPLETE_CANDIDATES} were incomplete)`)
+			: ""
+	}`;
+	const resultMsg = formatAsList(
+		[
+			`Injected ${chalk.bold.green(summary.INJECTED)}/${chalk.bold.white(summary.TOTAL)} torrents`,
+			summary.FULL_MATCHES &&
+				`${chalk.bold.green(summary.FULL_MATCHES)} were full matches`,
+			summary.PARTIAL_MATCHES &&
+				`${chalk.bold.yellow(summary.PARTIAL_MATCHES)} were partial matches`,
+			summary.INCOMPLETE_SEARCHEES &&
+				`${chalk.bold.yellow(summary.INCOMPLETE_SEARCHEES)} had incomplete sources`,
+			summary.ALREADY_EXISTS && incompleteMsg,
+			summary.BLOCKED &&
+				`${chalk.bold.yellow(summary.BLOCKED)} were possibly blocklisted`,
```

Comment:
Since it's checking all searchees, the blocklist may be a false positive. It shouldn't be a big deal though as it's only displayed if a match couldn't be found.